app/services/company.js

```diff
@@ -1,56 +1,90 @@
 const path = require('path');
 
-const companyFilesFolder = require('../../config.js').companyFilesFolder;
 const addressRepository = require('../repositories/address');
 const fieldOfActivityRepository = require('../repositories/fieldOfActivity');
 const fileRepository = require('../repositories/file');
+const companyValidator = require('../validators/company');
+const addressService = require('../services/address');
 const timetableService = require('./timetable');
 const fileService = require('./file');
 
 const error = require('./error');
 const CrudService = require('./crudService');
+const { COMPANY_FILES_FOLDER, ORIGIN } = require('../../config.js');
 
 class CompanyService extends CrudService {
-  async create(company, files) {
-    if (company.id !== undefined) {
-      error.throwValidationError('Invalid company format.');
+  async get(id) {
+    const company = await this.repository.get(id);
+    if (!company) {
+      error.throwNotFoundError('Company not found.');
     }
 
-    const companyExists = await this.repository.getByPropsNonParanoid({ name: company.name });
-    if (companyExists) {
-      error.throwValidationError('Company already exists.');
+    const { fieldOfActivityId, ...companyToReturn } = company;
+    const fieldOfActivity = await fieldOfActivityRepository
+      .get(fieldOfActivityId);
+    companyToReturn.business = fieldOfActivity.name;
+
+    const companyAddresses = await addressRepository.getAllByProps({
+      companyId: id,
+    });
+    if (companyAddresses && companyAddresses.length) {
+      const {
+        streetName,
+        streetNumber,
+        apartmentNumber,
+        city,
+        country,
+      } = companyAddresses[0];
+      companyToReturn.address = `${streetName} ${streetNumber}, ap. ${apartmentNumber}, ${city}, ${country}`;
```

Comment:
It is better to return the address object (model) and let the front end decide how it wants to display the address.